p/api/greader.php

```diff
@@ -476,73 +528,37 @@ function streamContents($path, $include_target, $start_time, $count, $order, $ex
 			break;
 	}
 
-	if (!empty($continuation)) {
+	if ($continuation != '') {
```

Comment:
So we're fine if it's FALSE or 0? Also, is $continuation trimmed anywhere or could you end up with something empty-adjacent like a space?

Also, like above, != or !==?

Reply:
Yes, loose comparison there. The test should reject empty string, false, 0, null, etc. and only keep substantial strings.
But good point with sanitising those parameters. Will do so.

Reply:
Ah, so != (loose) basically means exactly the same thing as `empty()`? I know that PHP plays a bit fast and loose with things like 1 and 0 vs true and false, but I didn't realize empty strings were included in being falsy. I thought any string was truthy. Apparently the string `'0'` is also falsy.

Reply:
Yes indeed. `empty()` is mostly relevant when working with potentially non-existing entries such as in a array, and it is similar to doing `!isset($t[1]) || $t[1] == ''`. In the case where we have a variable and not an index, there is little point of using `empty()` and it used to be slower (I am not sure whether there is any difference nowadays).

Reply:
I find `empty()` slightly clearer in the sense that I don't necessarily think of an empty string as falsey while I do think of it as empty, but that's just my problem I suppose. :-P

Python is much more consistent in counting empty arrays, empty strings, etc. as falsey than the likes of JavaScript or PHP. But in Ruby, for example, empty strings are truthy, and in a sense they are in Lua too. In POSIX and related shells you also have to be well aware to use checks like `-z` (unset or empty) and `-n` (set and non-empty).

Incidentally, can we get this kind of fun out of PHP?
```javascript
bla = '0'; if (bla && bla == false) console.log('/sigh')
/sigh
```

It's truthy (non-empty string) but at the same time '0' == false. Perhaps even more confusingly, you can get the same behavior out of `new Boolean(false)` (the existence of any object is inherently truthy in JS, but its value == false).

Reply:
I do not think there is anything in PHP that can evaluate positively `$bla && $bla == false`

Reply:
I didn't mean that specific scenario, just if there's something similarly crazy lurking in PHP that I don't know about. ;-)

Reply:
https://www.destroyallsoftware.com/talks/wat :-)

Reply:
Fun, but no PHP. Well, at least I got the last one right after the hint. I didn't really dare think not a number at first because that just seemed too logical. xD

Reply:
All right, I have one for you. Guess the output :-P

```php
<?php
echo ctype_digit(1234) ? 'true' : 'false', "\n";
echo ctype_digit(123) ? 'true' : 'false', "\n";
echo ctype_digit('123') ? 'true' : 'false', "\n";
```

Reply:
My intuitive guess would be false false true but I'm guessing there's a catch. Now to check… ;-)

Reply:
Close enough: true false true. :-)
> If an integer between -128 and 255 inclusive is provided, it is interpreted as the ASCII value of a single character (negative values have 256 added in order to allow characters in the Extended ASCII range). Any other integer is interpreted as a string containing the decimal digits of the integer.